modules/casper.js

```diff
@@ -705,7 +724,7 @@ Casper.prototype.evaluate = function evaluate(fn, context) {
         return utils.clone(this.page.evaluate(fn));
     } else if (arguments.length === 2) {
         // check for closure signature if it matches context
-        if (utils.isObject(context) && eval(fn).length === Object.keys(context).length) {
+        if (utils.isObject(context) && fn.length === Object.keys(context).length) {
```

Comment:
PhantomJS'  `evaluate()` accepts a string containing a function declaration as a first arg. This part of the code is to ensure we supports this use case.